### DISCUSSION TITLE
Keycloak uses samesite=None and failing in Pen test

### DISCUSSION THREAD

**Discussion by selvakumar-invicara** [keycloak/keycloak#22292] [Q&A]
I understand that Keycloak uses Samesite=None explicitly to work in SAML. I have situation where you can submit the form from cross domain.

For ex, Take the HTML content from the personal info form and save it in a html file. The form got successfully submitted and changed the profile information. So keycloak forms can be submitted from outside. I understand the form has csrf and works only in the session.

As per Pentest, they marked it as critical. My questions are
1. Is it possible to change samesite to lax from None without affecting IDP? If possible the how?  
2. Is there a standard to use samesite=none in SAML communications.
3. How to solve or How to make get succeed the pen test

**Comment by darius-m** [comment]
Please follow the [reporting a security vulnerability guide](https://github.com/keycloak/keycloak/security) if you believe this could be a security issue. Maybe @abstractj and @stianst could take a quick look and decide on the scope, since this has been in the open for a bit.

**Reply by vmuzikar** [reply]
+1
